tests/unit/device.ts

```diff
@@ -78,6 +82,7 @@ describe('Device', function() {
   beforeEach(() => {
     pstream = null;
     publisher = null;
+    updateAvailableDevicesStub = sinon.stub().returns(Promise.reject());
```

Comment:
If I recall correctly, I had issues when trying something similar to this because the `Promise.reject()` occurs in the closure of the `beforeEach` and not when the stub actually fires. But I could be misremembering...

Reply:
I did some test to verify this behavior. The rejection does happen when the stub fires. I know, confusing :)